runners/spark/src/main/java/org/apache/beam/runners/spark/translation/TransformTranslator.java

```diff
@@ -569,8 +569,8 @@ private static Partitioner getPartitioner(EvaluationContext context) {
     Long bundleSize =
         context.getSerializableOptions().get().as(SparkPipelineOptions.class).getBundleSize();
     return (bundleSize > 0)
-        ? null
-        : new HashPartitioner(context.getSparkContext().defaultParallelism());
+        ? new HashPartitioner(context.getSparkContext().defaultParallelism())
+        : null;
```

Comment:
Second thoughts, it was correct.

https://github.com/apache/beam/pull/6884/files#r246077919

This was in order to maintain old functionality (bundleSize == 0, which basically means to use predefined parallelism).

I think the old functionality doesn't make much sense as it doesn't scale with input data. I guess someone may use this in order to re-scale "downstream" stage, but there should be a better mechanism to do this.

Any thoughts? @timrobertson100 @kyle-winkelman

Reply:
I see. I reverted back the changes made to `getPartition`

Reply:
I agree that the old functionality seems strange, but I remember (when I had the logic backwards) that the performance tests for the spark runner were impacted. I think the impact was in streaming mode because if you don't use this HashPartitioner then it actually does a double shuffle of the data. I tried to clean this up but my I never finished getting PR #6511 merged.